curp/src/client.rs

```diff
@@ -443,16 +442,25 @@ where
                 }
             };
             debug!("shutdown request sent to {}", leader_id);
+
             let resp = match self
                 .get_connect(leader_id)
                 .unwrap_or_else(|| unreachable!("leader {leader_id} not found"))
                 .shutdown(
-                    ShutdownRequest::default(),
+                    ShutdownRequest::new(propose_id.clone(), self.cluster_version()),
                     *self.config.wait_synced_timeout(),
                 )
                 .await
             {
                 Ok(resp) => resp.into_inner(),
+                Err(RpcError::StatusError(status)) => {
+                    if matches!(status.code(), tonic::Code::FailedPrecondition) {
```

Comment:
The `Status::Code` has implemented the `Eq` trait. You can directly use "status.code() == tonic::Code::FailedPrecondition". It looks more concise.

Reply:
Done